### PR TITLE
Advanced user mangement vignette 4.38

### PR DESCRIPTION
**Pull request by nickdickinson**
Adding advanced user management vignette, updated reading of csv and xslx files to match modern expectations in the vignette and respective tidyverse dependencies in "suggests", added ability also to use roleAssignment() in addDatabaseUser() to align addDatabaseUser() and updateUserRole()

### CODE REVIEW

File: vignettes/advanced-user-management.Rmd [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+      },
+      error = function(e) {
+        warning(sprintf("Failed to add user %s with error: %s.", users[[i,"email"]], e))
+      }
+    )
+}
+
+```
+
+We will use the parameters may be defined through a "Name" column in a reference table and we will allow users to add the record "Name" instead of the record id in the reference table.
```
Comment (jamiewhths) [inline vignettes/advanced-user-management.Rmd:298]: I'm sure what this line is trying to say

File: vignettes/advanced-user-management.Rmd [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+
+First we will list the optional grants and get the grant resource label.
```
Comment (jamiewhths) [inline vignettes/advanced-user-management.Rmd:300]: Also seems an impromptu end

**Review by jamiewhths — APPROVED** [review]
Few lines to check here but otherwise looks good to go

File: vignettes/advanced-user-management.Rmd (hunk 1 above)
Comment (nickdickinson) [reply vignettes/advanced-user-management.Rmd]: I've done a big refactor and re-write for clarity and better code. It is much better and was my prep for the webinar.

File: vignettes/advanced-user-management.Rmd (hunk 2 above)
Comment (nickdickinson) [reply vignettes/advanced-user-management.Rmd]: Yes, now it's better and I've rearranged it a bit, with first generating a template CSV from the actual user list.

**Pull request merged**